### PR TITLE
[gui] DialogSeekbar should use g_application.GetPercentage() for progress info

### PR DESCRIPTION
**Pull request by phil65**
Forgot to PR this some time ago...
@FernetMenta

### CODE REVIEW

**Comment by FernetMenta** [comment]
thanks!
+1

**Comment by phil65** [comment]
jenkins build this please.

**Comment by phil65** [comment]
changed it to use MathUtils::round_int() instead of a static cast.
that way it aligns with https://github.com/xbmc/xbmc/blob/master/xbmc/GUIInfoManager.cpp#L6741 and fixes a visual glitch for seekbar (different values for player.progress vs value of dialogseekbar slider, control id 401)
Your choice if this is worth backporting, @FernetMenta 
Jenkins build this please.

**Comment by garbear** [comment]
I think we prefer std::roundf() as opposed to MathUtils because the latter is intended for pixel rounding in GUI code

**Comment by phil65** [comment]
@FernetMenta , @garbear 	
updated, and appended a 2nd commit to drop MathUtils in InfoManager.
Fine that way?
And do you want a backport?

**Comment by FernetMenta** [comment]
thanks. I would keep backports to a minimum of critical fixes but I am fine if you want to backport it.

jenkins build this please

**Comment by phil65** [comment]
jenkins build this please.

**Comment by phil65** [comment]
now using roundf(), round() and lround() + proper casting where appropriate to avoid warnings.
jenkins build this please.

File: xbmc/dialogs/GUIDialogSeekBar.cpp [hunk 1]
```diff
@@ -18,6 +18,8 @@
  *
  */
 
+#include <math.h>
```
Comment (phil65) [inline xbmc/dialogs/GUIDialogSeekBar.cpp]: Thx! Will fix.

**Comment by phil65** [comment]
jenkins build this please.

**Comment by garbear** [comment]
sorry, this should be `std::lrintf()` instead of `static_cast<int>(std::roundf())`

**Comment by FernetMenta** [comment]
lrintf still requires a cast to int

**Comment by garbear** [comment]
Documentation says `returns it as a value of type long int`, why would it still require a cast?

**Comment by FernetMenta** [comment]
> Documentation says returns it as a value of type long int, why would it still require a cast?

it gets assigned to parameter "value" which is of type int, not long int.

**Comment by garbear** [comment]
then you mean it requires an implicit cast instead of an explicit one, right?

**Comment by FernetMenta** [comment]
if you assign long to int without explicit cast you get a warning, don't you?

**Comment by garbear** [comment]
> if you assign long to int without explicit cast you get a warning, don't you?

On gcc you only get a warning if -Wconversion is specified, and I don't think it is for our project

**Comment by FernetMenta** [comment]
ok, then it's fine

**Comment by phil65** [comment]
updated. @garbear mind takin another look again?

**Comment by garbear** [comment]
looks good, thanks

**Comment by phil65** [comment]
jenkins build this please.

**Comment by phil65** [comment]
android std library doesnt seem to contain lrint(), is that correct? What should I use instead?

**Comment by phil65** [comment]
switched from <cmath> to <math.h>, that´s what we use everywhere else in code, too.

**Comment by phil65** [comment]
jenkins build this please.

**Comment by phil65** [comment]
jenkins build this please.

**Comment by phil65** [comment]
jenkins build this please.

**Comment by FernetMenta** [comment]
@phil65 afaik there is no lrintf on Android, just use lrint

**Comment by FernetMenta** [comment]
@phil65 great :) then shove it in

**Pull request merged**